presto-parser/src/main/java/com/facebook/presto/sql/tree/QualifiedName.java

```diff
@@ -45,10 +45,21 @@ public QualifiedName(String name)
 
     public QualifiedName(Iterable<String> parts)
     {
+        this(parts, true);
+    }
+
+    private QualifiedName(Iterable<String> parts, boolean toLower)
```

Comment:
Maybe a better way would be to have a private constructor that just takes the parts as they come (so it can be used in getPrefix), and a public static factory method that first downcases them before creating the object. It would require changing all call sites of `new QualifiedName(...)`, though.